src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+type State = {
+  x: string,
+};
+
+type Props = {
+  name: string,
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    x: new Date().toLocaleTimeString(),
+  };
+
+  timerId?: NodeJS.Timer;
+
+  componentDidMount() {
+    this.timerId = setInterval(() => {
+      const date: string = new Date().toLocaleTimeString();
+
+      this.setState({ x: date });
+
+      // eslint-disable-next-line
+      console.log(this.state.x);
+    }, 1000);
+  }
+
+  componentDidUpdate(oldName: Props) {
+    if (oldName.name !== this.props.name) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${oldName.name} to ${this.props.name}`);
+    }
+  }
```

Comment:
Check if the message appears in the console if you hide clock time
I can change the clock name when clock is hidden, but I don't see this message on the console